typescript/base/src/projects/ima/mainnet/MainnetImaProject.ts

```diff
@@ -1,17 +1,21 @@
+import { ContractAddressMap, MainContractAddress } from "../../../domain/types";
 import { ImaProject } from "../ImaProject";
 import { Instance } from "../../../instance";
 import { MainnetImaInstance } from "./MainnetImaInstance";
 
 export class MainnetImaProject<ContractType> extends
     ImaProject<ContractType> {
+    mainContractName = "MessageProxyForMainnet";
+
     getAbiFilename (version: string) {
         return `${this.metadata.name}-${version}-abi.json`;
     }
 
-    createInstance (address: string): Instance<ContractType> {
+    createInstance (target: MainContractAddress | ContractAddressMap)
```

Comment:
Maybe let it be `address` in assumption that `ContractAddressMap` is also an "address" in some sense?
For me `target` is too confusion because it's not clear what to pass.